src/Symfony/Component/Security/Core/User/CurrentUserProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Current User Provider.
+ *
+ * This provider gives you the current logged in user.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CurrentUserProvider
+{
+    /**
+     * @var TokenStorageInterface tokenStorage
+     */
+    private $tokenStorage;
+
+    /**
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(TokenStorageInterface $tokenStorage)
+    {
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Get a user from the Security Token Storage.
+     *
+     * @return mixed
+     *
+     * @see TokenInterface::getUser()
+     */
+    public function getUser()
+    {
+        if (null === $token = $this->tokenStorage->getToken()) {
+            return;
+        }
+
+        if (!is_object($user = $token->getUser())) {
```

Comment:
Moreover a user is not necessarily represented by an object, it depend on the implementation of token. (for example UsernamePasswordToken accept string representation to handle InMemoryUserProvider)